2_kata.py

```diff
@@ -0,0 +1,8 @@
+#https://www.codewars.com/kata/550554fd08b86f84fe000a58/train/python
+
+def in_array(a1, a2):  
+    #finding all possibles words
+    match = [word for word in a1 for second_words in a2 if word in second_words]
+    #removing duplicated, and sort alphabetical
+    return sorted(list(set(match)))
```

Comment:
Awesome handling of all the requirements in the task!

Just as an info, for you to know, the alternative solution:
```
  substrings = []
  for s1 in a1:
    for s2 in a2:
      if s1 in s2 and  s1 not in substrings:
        substrings.append(s1)
  substrings.sort()
  return substrings
```
The biggest difference here is that empty list `substrings` has to be predefined to be able to append things to it within the loops and the `if` statement already checks for duplicates, so in the end the `substrings` list only has to be sorted without other modifications,.